libvast/src/system/spawn_explorer.cpp

```diff
@@ -27,18 +29,67 @@
 #include <caf/local_actor.hpp>
 #include <caf/settings.hpp>
 
+#include <optional>
+
 using namespace std::chrono_literals;
 
 namespace vast::system {
 
+caf::error explorer_validate_args(const caf::settings& args) {
+  auto before_arg = caf::get_if<std::string>(&args, "explore.before");
+  auto after_arg = caf::get_if<std::string>(&args, "explore.after");
+  auto by_arg = caf::get_if(&args, "explore.by");
+  if (!before_arg && !after_arg && !by_arg)
+    return make_error(ec::invalid_configuration, "At least one of '--before', "
+                                                 "'--after', or '--by' must be "
+                                                 "present.");
+  vast::duration before = vast::duration{0s};
+  vast::duration after = vast::duration{0s};
+  if (before_arg) {
+    auto d = to<vast::duration>(*before_arg);
+    if (!d)
+      return make_error(ec::invalid_argument, "Could not parse", *before_arg,
+                        "as duration.");
+    before = *d;
+  }
+  if (after_arg) {
+    auto d = to<vast::duration>(*after_arg);
+    if (!d)
+      return make_error(ec::invalid_argument, "Could not parse", *after_arg,
+                        "as duration.");
+    after = *d;
+  }
+  if (!by_arg) {
+    if (before <= 0s && after <= 0s)
+      return make_error(ec::invalid_argument,
+                        "At least one of '--before' or '--after' must be "
+                        "greater than 0 "
+                        "if no spatial constraint was specified.");
+  }
+  return caf::none;
+}
+
 maybe_actor spawn_explorer(node_actor* self, spawn_arguments& args) {
   if (!args.empty())
     return unexpected_arguments(args);
-  auto before
-    = get_or(args.invocation.options, "explore.before", vast::duration{0s});
-  auto after
-    = get_or(args.invocation.options, "explore.after", vast::duration{0s});
-  auto expl = self->spawn(explorer, self, before, after);
+  if (auto error = explorer_validate_args(args.invocation.options))
+    return error;
+  auto maybe_parse
+    = [](caf::optional<std::string>&& str) -> std::optional<vast::duration> {
+    if (!str)
+      return std::nullopt;
+    auto parsed = to<vast::duration>(*str);
+    if (!parsed)
+      return std::nullopt;
+    return *parsed;
+  };
+  auto before = maybe_parse(
+    caf::get_if<std::string>(&args.invocation.options, "explore.before"));
+  auto after = maybe_parse(
+    caf::get_if<std::string>(&args.invocation.options, "explore.after"));
+  std::optional<std::string> by
+    = to_std(caf::get_if<std::string>(&args.invocation.options, "explore.by"));
+  auto expl = self->spawn(explorer, self, before, after, by);
   return expl;
```

Comment:
```suggestion
  return self->spawn(explorer, self, before, after, by);
```

Reply:
Resolving as I don't want to re-run CI for this, but I changed it in my working tree and promise to add it into the review for adding limits to `vast explore` :crossed_fingers: